app/femr/business/services/system/EncounterService.java

```diff
@@ -535,4 +536,47 @@ private Integer getTabFieldIdByTabFieldName(List<? extends ITabField> tabFields,
 
         return tabFieldId;
     }
+//troubleshooting
+
+    public ServiceResponse<List<PatientEncounterItem>> returnCurrentDayPatientEncounters()
+    {
+        ServiceResponse<List<PatientEncounterItem>> response = new ServiceResponse<>();
+        List<PatientEncounterItem> patientEncounterItems = new ArrayList<>();
+        //gets all patients encounters
+        ExpressionList<PatientEncounter> query = QueryProvider.getPatientEncounterQuery()
+              .where()
+                .ge("id",0);
```

Comment:
It looks like you are going to the database to grab ALL patient encounters then looping over each one to check if it is equal to the current date. In a real system this would pull just over 5,500 encounters  today which shouldn't be an issue, but performance will be impacted as the number of encounters grow

You should be able to use the same logic on dates to reduce the number of encounters being fetched from the database. Instead of .ge("id",0), you should be able to do something like: 

.where()
.ge("date_of_triage_visit", 12:00:00am today)
.le("date_of_triage_visit", 12:00:00am tomorrow)

and you'll automatically get back the encounters you are looking for without the for loop